### ISSUE TITLE
Instant Uploads never started - ANDROID 7

### ISSUE THREAD

**Issue by kigero** [owncloud/android#1829]
### Actual behaviour

With Instant Upload turned on, no pictures or videos are automatically uploaded, with or without the "wifi only" toggle on.  After reading the related issues, it seems like generally Instant Upload works on Android 7, but some specific phones are having issues.
### Expected behaviour

Pictures and videos should automatically upload.
### Steps to reproduce
1. Enable instant upload in the settings.  Set the appropriate folder for pictures (tried both internal and external SD card).
2. Take a picture.
3. The picture is not uploaded - and does not show in the Uploads section.

Can this problem be reproduced with the official owncloud server?
(url: https://demo.owncloud.org, user: test, password: test)
Yes
### Environment data

Android version: 7

Device model: LG v20

Stock or customized system: Stock

ownCloud app version: 2.1.2 (built from master branch on Oct 22).

ownCloud server version: 8.1
#### Web server error log

No relevant logs found.
#### ownCloud log (data/owncloud.log)

No logs found.

**Comment by davivel** [comment]
Hi, @kigero .

What camera app are you using? 

What are the exact paths you are setting for the camera folder?

Was the device originally shipped with Android 7, or was it upgraded from Android 6?

Thanks in advance.

**Comment by kigero** [comment]
I"m using the default camera app (com.lge.camera).

The path that I am currently using for the SD card is: /storage/6264-3033/DCIM/Camera.  The path that is used by the internal card - and which Owncloud originally picked up automatically - is: /sdcard/DCIM/Camera.  I've tried both with no success.

It originally shipped with Android 7.

Thanks!

**Comment by piperfect** [comment]
I am having similar results with the LG V20 without using an external SD card.  The default camera path after install is /storage/emulated/0/DCIM/Camera

**Comment by davivel** [comment]
Thank you both. We need to fix this, but version 2.2.0 needs to go before the end of the year and it's currently too loaded. I'm sorry, will have to wait to the next one.

**Comment by kigero** [comment]
I did a bit of digging this morning and found a few interesting notes that might be useful when it comes time to take a look at this:

1. The InstantUploadsObserver  just doesn't seem to be triggering at all on that folder, but it does seem to be configured and started properly - I'm not seeing any errors in the log.
2. The InstantUploadBroadcastReceiver is configured to listen for `android.hardware.action.NEW_PICTURE` intents, but it seems that the v20 is sending the `com.lge.camera.action.ACTION_NEW_PICTURE` intent - at least that's what shows up in the logs when I take a picture.  I tried adding that event to the intent-filter, but ended up with some permission issues that I couldn't reconcile:

`11-23 09:39:41.111: W/BroadcastQueue(3369): Permission Denial: receiving Intent { act=com.lge.camera.action.ACTION_NEW_PICTURE dat=content://media/external/images/media/15523 flg=0x10 } to com.owncloud.android/.broadcastreceivers.InstantUploadBroadcastReceiver requires com.lge.permission.LGE_CAMERA_ACTION due to sender com.lge.camera (uid 10076)`

I have no idea if that is relevant or not, but thought I'd share it.

**Comment by houseofbugs** [comment]
I have the LG V20 as well and I am having the same issue. When is a fix expected? This is killing the platform for me. :( Thanks!

**Comment by andreaphsz** [comment]
same for sony xperia x compact andorid 7.0

**Comment by ToonOwncloud** [comment]
Hi, just to confirm: since a week I own an LG V20 and I ran into the same problem.
I can't get instant upload working, while it was working fine on my previous phone with cyanogenmod.
I tried with internal as well as external (microSD) storage.
The version of the ownCloud client app is 2.2.0

**Comment by smmilani** [comment]
I have the same problem with Galaxy S7 Edge running android 7.0 with owncloud app 2.2.0
It worked fine before I upgraded the phone from 6.0.1 to 7.0

**Comment by jesmrec** [comment]
Have anyone check if the problem happens using a different camera app? there are some free camera apps in Google Play like OpenCamera or CameraMX. It would be very interesting to know if the source problem is related with the camera app in these kind of devices using Android 7.0

**Comment by davivel** [comment]
> I did a bit of digging this morning and found a few interesting notes that might be useful 

Thanks, @kigero, we are aware of that problem. We provided a separate method to observe for new pictures with version 2.1.1 of the app, and then again with 2.1.2. That's why the option 'Camera folder' was added.

Unfortunately this seems not good enough in some devices yet.

Let me rephrase the title. We'd prefer to have a single issue for all Android 7 devices. 

Unfortunately, at the moment I strongly doubt we can fix this for version 2.3.0 (it's quite loaded). For the moment I'll leave the milestone there, but chances are we move it in the future to 2.4.0.

**Comment by seisen** [comment]
i installed opencamera,  set destination folder to internal storage,  outside of DCIM and InstantUpload got triggered.

**Comment by songpr** [comment]
Samsung Galaxy S7 not work too.
Environment data

Android version: 7

Device model: Galaxy S7

Stock or customized system: Stock

ownCloud app version: 2.2.0

ownCloud server version: 9.0

**Comment by jesmrec** [comment]
@songpr which camera app?

**Comment by dauuser** [comment]
i have the same problem with the galaxy S7, with the samsung camera app. I have reset the phone after update to android 7. my owncloud server is the 9.0.
My wish would be to fix it quickly :-)

**Comment by jensd0e** [comment]
In the meantime, use nextcloud *latest beta* app, where the issue is fixed. Maybe someone can borrow that fix and port it over to owncloud client app?

**Comment by Trini2906** [comment]
I have the same problem with my sony xperia z5 compact after upgrading to Android 7.0. I'm using the default camera app and a path to my sd-card (/storage/7235-E2AB/DCIM/102ANDRO).
App version: 2.2.0
Server version 9.1.4

**Comment by dauuser** [comment]
Hi jensd0e,
I have update the Server to 9.1.4 stable. 
have I understood that correctly, i can use the nextcoud app with owncloud server?

**Comment by jensd0e** [comment]
Yes. But you need to use the beta app from GitHub.

**Comment by piperfect** [comment]
Thank You jensd0e! 
The nextcloud beta works nicely

**Comment by jaescribano** [comment]
Same server with android 6 - sync is working / with android 7 - sync not working.
Manual upload works fine, so it's not  a matter of permissions. Checked app access, disabled and re enabled... no luck. it just does not sync.
regards

**Comment by moulip** [comment]
Hi,
Actual behaviour

Since I have updated my S7 Edge to Nougat 7.0, the auto-instant upload does not work any longer.
I mean nothing happens not even the cloud icon appearing and throwing an error, nothing.
I have tried reinstalling the app, restoring the phone to factory settings, try to check on wifi only or no, nothing is working. That's a shame considering the role of a cloud application. But it works well on my other phone a Huawei P9 Lite running Android 6.0.1.
Steps to reproduce

    Setup the app to autoupload photos and videos
    take a photo
    check that nothing is happening

Can this problem be reproduced with the official owncloud server?
(url: https://demo.owncloud.org, user: test, password: test)

yes
Environment data

Android version: 7

Device model: Samsung S7 EDGE

Stock or customized system: STOCK

ownCloud app version: 2.3.0

ownCloud server version: 8.2.10

Camera App : Samsung

**Comment by jaescribano** [comment]
hi! thanks for your answer
Checked noting is happening.
Looks to me there is an issue with devices running nougat.
thanks

2017-03-21 11:59 GMT+01:00 moulip <notifications@github.com>:

**Comment by sastorsl** [comment]
Same issue on my Samsung Galaxy S7 after updating to Nougat.

Right now the app crashes - and started doing so after I updated the owncloud settings to point to the new camera folder. Even after uninstalling/installing Owncloud, deleting settings, the app still crashes. All the time and every time it is started.

After re-installing, I start, get a crash, asked about giving feedback/closing, pressing feedback, then back, I get into adding a new server. It connects to my server, and I type my user/pass, and then it crashes again.

**Comment by sr222** [comment]
Same here: no upload since Android 7. My phone is a Samsung Galaxy S7. After I read your comments, I tried several camera app: it doesn't work either. It was working just fine before Android 7 upgrade. This is VERY annoying. And my owncloud app is 2.3.0 so this was absolutely not fixed with last version...

**Comment by dauuser** [comment]
Hi,

i have install the **nexcloud** app from the Google Play Store yesterday. Now it's running fine.
I use a Samsung Galaxy 7 with Android 7.

Best regards

**Comment by sr222** [comment]
@dauuser Thanks a lot. It seems to work perfectly !

**Comment by davivel** [comment]
Sorry, but we need to drop issues from milestone 2.4.0 to get it out next to OC 10 server.

This time we'll move topics to ```backlog``` instead of directly to next release, so that we can schedule 2.5.0 properly.

**Comment by livingclueless** [comment]
Hi Folks,

had same issue with the S7 too, very frustrating, but easy to solve, just changed the camera picture folder to device. Seems not to work with the SD-Card.

Not the best, but since owncloud doesn't work with SD-Cards at all... i don't mind and enjoy my owncloud working again.

have a nice day

**Comment by kdslkdsaldsal** [comment]
> but easy to solve, just changed the camera picture folder to device. Seems not to work with the SD-Card.

Nice, this did the trick here as well. Thanks

**Comment by houseofbugs** [comment]
That isn't a fix and it doesn't work on my phone.

**Comment by kdslkdsaldsal** [comment]
> That isn't a fix and it doesn't work on my phone.

No one said that this is a fix 😉 .

**Comment by moulip** [comment]
On Mon, 2017-04-24 at 00:06 -0700, houseofbugs wrote:
> That isn't a fix and it doesn't work on my phone.
> 
> 

I had thought about that idea too. This does not work on my Huawei Mate
9 neither.

**Comment by nejkica** [comment]
Note 3, cyanogenmod 14.1, same problem. 

When using cameraMX it works again. Just thought I would share...

**Comment by kdslkdsaldsal** [comment]
@moulip @nejkica @houseofbugs If the workaround doesn't work for you then it might be also possible that you're just having an additional issue not related to Android 7? There are quite some open issues about instant upload available here:

**Comment by nejkica** [comment]
As I have browsed through the link you sent me, I then tried to use
CameraMX app. And now it works.
I like the idea about the -  listen for event "new picture". It seems
likely. Is it possible that other apps generate this event differently?
Anyway it works now...

Thank you and best regards, Jernej

25. apr. 2017 14:09 je oseba "kdslkdsaldsal" <notifications@github.com>
napisala:

**Comment by janipewter** [comment]
Galaxy S7 (Android 7.0) user here with same problem. Camera upload never starts when using the native camera app. Tried internal and microSD storage. I don't fancy using a third party camera app either :-(

**Comment by roots84** [comment]
Use Nextcloud client for the time being. Works beautifully on Galaxy S7 with my ownCloud instance

**Comment by moulip** [comment]
Hello, I have just given Nextcloud a try and I confirm it works like a charm on my Mate 9.And it's free :-)                                                                                                                                                                         Envoyé depuis BlackBerry Hub pour Android                                                                                                                                                                            De: notifications@github.comEnvoyé: 8 mai 2017 11:09 AMÀ: android@noreply.github.comRépondre à: reply@reply.github.comCc: julien@technoaddict.fr; mention@noreply.github.comObjet: Re: [owncloud/android] Instant Uploads never started - ANDROID 7 (#1829)  Use Nextcloud client for the time being. Works beautifully on Galaxy S7 with my ownCloud instance

**Comment by davivel** [comment]
Let's go for this one, @jesmrec , @davigonz .
 
First iteration, minimum to fix: subscription to update of ```MediaContentProvider```  as recommended by Google.

**Comment by nasli** [comment]
First approach that improves the behavior done. 

New beta app will be available soon in fdroid so you can give us feedback.

**Comment by jesmrec** [comment]
ownCloud Beta already available in F-droid. All feedback in apps@owncloud.com or here will be appreciated. Included the first approach to the fix the broken instant uploads.

**Comment by davivel** [comment]
Hi all @kigero, @piperfect, @houseofbugs, @andreaphsz, @ToonOwncloud, @smmilani, @seisen, @songpr, @dauuser, @Trini2906, @jaescribano, @moulip, @sastorsl, @sr222, @kdslkdsaldsal, @houseofbugs, @nejkica, @janipewter, @livingclueless, @jensd0e and @roots84 .

Let me invite you to try the last release of the ownCloud Beta app that includes a candidate fix for your issue with instant uploads. This beta app can be installed in your devices besides of the regular ownCloud app, and both will work independently.

You will notice that uploads do not start immediately, we set an interval of 5 minutes to poll for new pictures in the camera folder. Please, try this fix and tell us your impressions about it. There is room for many improvements yet, and any feedback from you will be very appreciated.

For the moment you can find ownCloud Beta in F-Droid. You will need to install the F-Droid client in your device from https://f-droid.org/FDroid.apk, and then search "ownCloud Beta" in it.


Thank you so much.

**Comment by kdslkdsaldsal** [comment]
> You will need to install the F-Droid client in your device from https://f-droid.org/FDroid.apk, and then search "ownCloud Beta" in it.

For testing purposes the ownCloud Beta client APK is also available directly at:

https://f-droid.org/packages/com.owncloud.android.beta/
-> https://f-droid.org/repo/com.owncloud.android.beta_20400000.apk

**Comment by kigero** [comment]
I've been using this for a few days now, and it seems to be _partially_ working.  What I've observed is that pictures and videos that I've taken over the last few days have ended up being uploaded, only not necessarily when I would expect, and not all pictures either.  For example:

* I took several pictures last night that I found had been uploaded at some point overnight.  
* I took a picture earlier this morning and it didn't upload (didn't make it into the Uploads section of the app either).  
* I just took a second picture and it _immediately_ went into Uploads and uploaded (I'm guessing this just hit the right time in the polling interval).
* I then took a third and fourth picture, and neither of them did anything right away - no entry in Uploads - but after a few minutes were put into Uploads and uploaded.

All of this was done while connected to wifi.  So definitely some progress, but it still seems to be missing some files.

**Comment by kigero** [comment]
I've been using this for a few days now, and it seems to be _partially_ working.  What I've observed is that pictures and videos that I've taken over the last few days have ended up being uploaded, only not necessarily when I would expect, and not all pictures either.  For example:

* I took several pictures last night that I found had been uploaded at some point overnight.  
* I took a picture earlier this morning and it didn't upload (didn't make it into the Uploads section of the app either).  
* I just took a second picture and it _immediately_ went into Uploads and uploaded (I'm guessing this just hit the right time in the polling interval).
* I then took a third and fourth picture, and neither of them did anything right away - no entry in Uploads - but after a few minutes were put into Uploads and uploaded.

All of this was done while connected to wifi.  So definitely some progress, but it still seems to be missing some files.

**Comment by davivel** [comment]
Thanks, @kigero.

> I took several pictures last night that I found had been uploaded at some point overnight.

We expected this.

> I took a picture earlier this morning and it didn't upload (didn't make it into the Uploads section of the app either).

We didn't expect this; or, to be honest, we didn't expect it happened so easily. More about this below.

> I just took a second picture and it immediately went into Uploads and uploaded (I'm guessing this just hit the right time in the polling interval).

You guess correctly, we expected this.

> I then took a third and fourth picture, and neither of them did anything right away - no entry in Uploads - but after a few minutes were put into Uploads and uploaded.

We expected this too.

Thanks for all the details, it's useful for us knowing that you check the 'Uploads' view to know if pictures have been detected or not. With the current code you will see no change in the view until the pictures are scanned by the periodic polling, but we need to handle the fact that users expect to see something there.

About the lost picture: that is very bad. During our analysis we found two sources of failure for instant uploads. The first one was that the ```FileObserver``` class in Android API is not trusted in many devices from Android 6, specially in non-Google devices. We fixed that in this beta version, and that's way absolutely no picture was detected in your device, and now some of them are detected.

The second problem is more tricky. The observers we register for changes in the camera folder "live" in a background service that we try to keep alive forever. If that service is killed for any reason, the observers "die" and pictures are not detected anymore until the service is restarted. The restart is done when the app is opened again, or when the device is rebooted. 

In the past, the death of the service didn't happen often. But from Android 4.4, Google is **silently** killing "immortal" services more and more often with each Android version. With the incoming Android 8 Google finally is documenting that services in background will not be allowed to live more than a limited time. 

We expected that the fix in the current beta was good enough for a while, but if you are experiencing "picture loss" in a period of 12 hours, that means that the service is dying much more frequently than we expected.

We need to go for a total rewrite of the feature ASAP. Say good bye to instant uploads, it's the time for automated uploads (instant-no-more).

cc @michaelstingl

**Comment by kigero** [comment]
Just to add a bit more information, out of 147 pictures and videos taken over the last week or so, about 20-25 did not get automatically uploaded.  Looking at the ones that didn't make it, about half of them are in the Uploads section but seem to be stuck on "waiting for wifi connectivity".  These were taken yesterday off of WiFi, and once I was connected to WiFi again it looks like some of them uploaded properly while some of them didn't - I don't see any pattern in the ones that didn't upload.

**Comment by jensd0e** [comment]
Just a note: I just came back from a short holiday trip: nextcloud [dev] didn't work flawlessly either. Sometimes, photos didn't get uploaded. Had to cycle wifi to make it work again. So this does not seem to be a good starting point neither.

The only solution that did work flawlessly (but on wifi only), was Google Photos.

**Comment by d98ama** [comment]
Just a comment on the problem with some photos never making it to the "upload section".

I temporary switched to the Nextcould app to fix the auto upload on my Samsung Galaxy S7. In the beginning I had a similar problem as @kigero, some photos never appeared in the uploads section in the Nextcloud app. 
After excluding the app from "Power Monitoring" (Settings -> Device maintenance -> Battery -> Unmonitored apps), it seems to work better. I don't think that it has missed a single photo since I did that change.

Maybe this can make things a bit better for the owncloud app as well (if you are using a Samsung phone that is).

**Comment by conlledo** [comment]
Hi, anybody that have the problem with the S7 with Nougat, I updated all pending updates from T-Mobile via OTA and ownCloud started working. It was yesterday 07/30/2017.

**Comment by jreiter1975** [comment]
Nope, still doesn't work on a VZW LG G5 with Android 7.0 and June 1st security patch.

**Comment by davivel** [comment]
Thank everybody for your feedback, it's really useful. 

This point by @d98ama is really good:

> After excluding the app from "Power Monitoring" (Settings -> Device maintenance -> Battery -> Unmonitored apps), it seems to work better. 

This should prevent that the system kills the observer service so often. Let me invite you all to try it, and observe if the impact on your battery is really meaningful.

We still couldn't rewrite the feature as intended, we need to complete and release support for OAuth2 log-in first. There will be a version 2.5.0 of the app released soon that includes the improvements already in the beta version. But for a real fix, I'm afraid it will have to wait for version 2.5.1.

Thank you so much for your patience.

**Comment by erwinfolmer** [comment]
Maybe it is already known, then please ignore this post:
But last week I updated my Samsung A5 to Android Nougat, and since then it does not upload any photos anymore (while before it worked perfectly). Tried different settings, but no results so far. It looks the same as mentioned before in this tread...so you can add Samsung A5 also to the list.

**Comment by jesmrec** [comment]
@erwinfolmer could you check with the 2.5.0beta in FDroid/Google Play

**Comment by erwinfolmer** [comment]
@jesmrec Not sure if I am able to do so; I am using the SurfDrive app, and if I understand correctly this is a skin on owncloud?

**Comment by michaelstingl** [comment]
@erwinfolmer You could test the beta with https://demo.owncloud.org/ (demo:demo)

**Comment by erwinfolmer** [comment]
@michaelstingl I did, but no success at all....it doesn't upload the pictures...

**Comment by michaelstingl** [comment]
@erwinfolmer Thank you very much for your feedback. We plan a 2.5.1 that hopefully fixes your issue too.

**Comment by davivel** [comment]
> @jesmrec Not sure if I am able to do so; I am using the SurfDrive app, and if I understand correctly this is a skin on owncloud?

You can install both apps in the same device. They work independently, though if you use the same server account in both, probably you will prefer to uninstall the beta from the device after some testing.

**Comment by erwinfolmer** [comment]
Hi, just to add to the issue: if I change the camera uploads from the SDcard to the internal phone memory, than instantupload works again! (no matter what the settings are within owncloud; even when these settings are set to a directory from the SDcard, it will upload from internal memory; it seems that Owncloud  ignores this folder settings...)

**Comment by Diamondog** [comment]
As of today 9/22/17 OWNCLOUD app on my Android Note 8 still does not work. I did get NEXTCLOUD and that does work , But not sure why owncloud has not fixed this problem. Just happy that this forum was here to help me with the problem. Thank you!

**Comment by michaelstingl** [comment]
@Diamondog The ownCloud app 2.4.0 from May don't has the fix. The fix is only available in the ownCloud 2.5.0-beta2 app available also on Play Store or F-Droid. A final 2.5.0 release with the fix will be available in the Play Store very soon.

**Comment by fishstew** [comment]
I have tried the beta app and sadly the instant upload feature is not working.

Make: Sony
Model: Xperia XZ (g8141)
Android Version: 7.1.1
Owncloud Client version: 2.5.0-beta2 
Camera App: Inbuilt one.

**Comment by jesmrec** [comment]
@fishstew are the pics not uploaded after 5 minutes?

**Comment by fishstew** [comment]
@jesmrec 
Just testing now.

Photo taken: 13:06 
Checked: 13:13
Status: Fail - nothing uploaded

**Comment by jesmrec** [comment]
if you can ship us logs (you can get them in settings view), it could be helpful. Post here or sent them to apps@owncloud.com. Thanks @fishstew

**Comment by fishstew** [comment]
@jesmrec there you go :-)

[currentLog.txt](https://github.com/owncloud/android/files/1348886/currentLog.txt)

**Comment by jesmrec** [comment]
Feature already released in 2.5.0 with other improvements. Please follow up in https://github.com/owncloud/android/issues/2028 and give us there your feedback.

Thanks to all contributors here.

**Comment by davigonz** [comment]
Hi @kigero , @piperfect , @andreaphsz , @ToonOwncloud , @smmilani , @seisen , @songpr , @dauuser , @Trini2906 , @jaescribano , @moulip , @sastorsl , @sr222 , @livingclueless , @nejkica , @janipewter , @roots84 , @d98ama , @conlledo , @jreiter1975 , @erwinfolmer , @Diamondog , @fishstew ,

we released yesterday the 2.6.0 Android app with the new **Camera uploads** feature, which will replace Instant Uploads and **will upload automatically your pictures and videos in 15 minutes at most**, focusing on reliability instead of immediacy and avoiding battery draining caused by excessive checking of the folder.

I invite you to try it out and share with us your impressions, thank you.

**Comment by sciurius** [comment]
Running 2.7.0 and it seems to work.
A preference setting for the camera path and the scan interval would be appreciated.

**Comment by dahamsta** [comment]
There's already a setting for camera path.

**Comment by sciurius** [comment]
I stand corrected. Thanks.

**Comment by dahamsta** [comment]
NP

**Comment by jesmrec** [comment]
The scan interval depends directly on the Android system and can not be controlled directly by the app.

**Comment by Diamondog** [comment]
/storang/emulated/0/dcim/camera

I do not see the interval setting on the beta or it is not showing at 
this time. BUT I do remember one somewhere before on your app. Maybe I 
just can't find it at this time. Where are is it located?

**Comment by nzjc** [comment]
Now with 2.7.0 I still have the same problem as I did back in 2016... I am running a Xiaomi 5S and while 'instant' upload did used to work periodically, Camera Upload doesn't seem to have worked ever. I wonder if my OS is just not doing the scan at all..

However, if I change to Open Camera and set a different folder (one level up from ../Camera in DCIM) - it does work just fine. Could this just be a problem with my Xiaomi version of Android? Seems so.

**Comment by jesmrec** [comment]
some devices adds a customization layer over the Android. If you want to help us to research which the problem with your device is, you can send a mail to apps@owncloud.com and there, we will give you more details about the info we need or the ways we can suggest you to ship us related info.
